coalib/parsing/TomlConfParser.py

```diff
@@ -0,0 +1,255 @@
+                 to_append: bool = False,
+                 ):
+        self.original_value = original_value
+        super(TomlSetting, self).__init__(
+            key,
+            value,
+            origin,
+            strip_whitespaces,
+            list_delimiters,
+            from_cli,
+            remove_empty_iter_elements,
+            to_append)
+
+
+class TomlConfParser:
+
+    def __init__(self, remove_empty_iter_elements=True):
+
+        self.sections = None
+        self.data = None
+        self.__rand_helper = None
+        self.__init_sections()
+        self.__remove_empty_iter_elements = remove_empty_iter_elements
+
+    def parse(self, input_data, overwrite=False):
+        """
+        Parses the input and adds the new data to the existing.
+
+        :param input_data: The filename to parse from.
+        :param overwrite:  If True, wipes all existing Settings inside this
+                           instance and adds only the newly parsed ones. If
+                           False, adds the newly parsed data to the existing
+                           one (and overwrites already existing keys with the
+                           newly parsed values).
+        :return:           A dictionary with (lowercase) section names as keys
+                           and their Setting objects as values.
+        """
+
+        if os.path.isdir(input_data):
+            input_data = os.path.join(input_data, Constants.local_coafile_toml)
+
+        if overwrite:
+            self.__init_sections()
+
+        with open(input_data, 'r') as file:
+            self.data = tomlkit.parse(file.read())
+
+        self.data = self.data.body
+
+        for item in self.data:
+            self.generate_section(item, input_data)
+        return self.sections
+
+    def get_section(self, name, create_if_not_exists=False):
+        """
+        Returns or creates a section with given name
+
+        :param name: The name of the section
+        :param create_if_not_exists: create a section if it does not exist
+        :return: Section of given name
+        """
+        key = self.__refine_key(name)
+        sec = self.sections.get(key, None)
+        if sec is not None:
+            return sec
+
+        if not create_if_not_exists:
+            raise IndexError
+
+        retval = self.sections[key] = Section(str(name))
+        return retval
+
+    @staticmethod
+    def __refine_key(key):
+        return str(key).lower().strip()
+
+    def generate_section(self, item, origin):
+        """
+        Generates section
+
+        :param item: Configuration group
+        :param origin: The file from which the configuration originated
+        """
+
+        section_name = item[0]
+        section_content = item[1]
+        appends = []
+
+        # Handle Default section
+        if not isinstance(section_content, Table):
+            original_value = section_content
+            current_section = self.get_section('default', True)
```

Comment:
in your coafile.toml the section name id Default, so even if the deafult section is there it will create a section name `default`

Reply:
This is for settings at top of the file without any sections. These settings which do not belong to any sections will be added to the default section.